src/PerformanceTable.cpp

```diff
@@ -1,49 +1,306 @@
+                  perf_vect[i].getId()) != perf_id_vect.end()) {
+      throw std::invalid_argument("Each performance must have different ids.");
+    }
+    perf_id_vect.push_back(perf_vect[i].getId());
+
+    // ensure all the performance are based on the same set of criterion
+    if (perf_vect[i].getCriterionIds() != crit_vect) {
+      throw std::invalid_argument("Each performance must be based on the same "
+                                  "set of criterion, in the same order.");
+    }
+    pt_.push_back(Performance(perf_vect[i]).getPerf());
   }
 }
 
 PerformanceTable::PerformanceTable(std::string prefix, int nb_of_perfs,
                                    Criteria crits) {
   for (int i = 0; i < nb_of_perfs; i++) {
-    pt_.push_back(Performance(prefix + std::to_string(i), crits));
+    pt_.push_back(Performance(prefix + std::to_string(i), crits).getPerf());
   }
 }
 
 PerformanceTable::PerformanceTable(const PerformanceTable &perfs) {
   // deep copy
-  for (int i = 0; i < perfs.pt_.size(); i++) {
-    pt_.push_back(Performance(perfs.pt_[i]));
+  for (std::vector<Perf> p : perfs.pt_) {
+    std::vector<Perf> perf_vect;
+    for (Perf perf : p) {
+      perf_vect.push_back(Perf(perf));
+    }
+    pt_.push_back(perf_vect);
   }
 }
 
 PerformanceTable::~PerformanceTable() {}
 
 std::ostream &operator<<(std::ostream &out, const PerformanceTable &perfs) {
   out << "PerformanceTable(";
-  for (Performance perf : perfs.pt_) {
-    out << perf << ", ";
+  for (std::vector<Perf> p : perfs.pt_) {
+    out << "Performance(";
+    for (Perf perf : p) {
+      out << perf << ", ";
+    }
+    out << "), ";
   }
   out << ")";
   return out;
 }
 
-Performance PerformanceTable::operator[](std::string name) const {
-  for (Performance p : pt_) {
-    if (p.getId() == name) {
-      return p;
+std::vector<Perf> PerformanceTable::operator[](std::string name) const {
+  // supposing the pt is consistent:
+  // if in mode alt, each row contains 1 and only 1 (alternative or profile)
+  // if in mode crit, each row contains 1 and only 1 criterion
+  if (mode_ == "alt") {
+    for (std::vector<Perf> p : pt_) {
+      if (p[0].getName() == name) {
+        return p;
+      }
+    }
+    throw std::invalid_argument("Row not found in performance table");
+  } else if (mode_ == "crit") {
+    for (std::vector<Perf> p : pt_) {
+      if (p[0].getCrit() == name) {
+        return p;
+      }
     }
+    throw std::invalid_argument("Row not found in performance table");
+  } else {
+    throw std::domain_error("Performance table mode corrupted.");
   }
-  throw std::invalid_argument("Row not found in performance table");
 }
 
-std::vector<Performance> PerformanceTable::getPerformanceTable() const {
+void PerformanceTable::generateRandomPerfValues(unsigned long int seed) {
+  srand(seed);
+  for (std::vector<Perf> &pv : pt_) {
+    for (Perf &p : pv) {
+      p.setValue((float)rand() / RAND_MAX);
+    }
+  }
+  sorted_ = false;
+}
+
```

Comment:
should we use a have a lower and upper bound limit for the values specified ? 
ie by default its between 0 and 1 but you can give btwn 1 and 4 for example ?

Reply:
Yeah we could add this, but aren't the values supposed to be in [0,1] ?

Reply:
I think they can be as big as one want but for optimization purpose the data probably needs to be normalised

Reply:
I see, we will need to add these normalization functions! I will add some bound limit args here to be more flexible